qa/workunits/rbd/rbd-ggate.sh

```diff
@@ -67,7 +82,8 @@ cleanup()
     then
 	_sudo rbd-ggate unmap ${DEV}
     fi
-    ceph osd pool delete ${POOL} ${POOL} --yes-i-really-really-mean-it
+    # sometimes this fails, but it is safe to ignore that
+    ceph osd pool delete ${POOL} ${POOL} --yes-i-really-really-mean-it || true
```

Comment:
@wjwithagen Is it necessary? Note `set +e` at the beginning of the `cleanup` function, which was added exactly for cases like this.

Reply:
@trociny 
If we do then I have to start figuring out why 50% of the time on different platforms the `pool delete` fails... It is not the purpose of this test to test `pool delete`, so I'd rather not stress on it.
Pool is being killed anayways right after, but the exit value of rbd-ggate.sh seems to be the result of the last executed program. Which can/could be an error from `pool delete`
Can also put an `exit 0` there

Come to think of it, `cleanup()` is always called, good or bad result:
`trap cleanup INT TERM EXIT`
So how to differentiate regular cleanup, of cleanup on error?

And ATM I have not enough time to spent on Ceph as it is, so I'd rather find some time to work on ceph-volume-zfs. But then I get distracted by keeping up, and another weekend had gone.

Reply:
I agree that we don't need to test pool delete here, but my point is that the current version of the script should not exit with non-zero status if the pool deletion fails in cleanup. I just tried this and it worked as I expected:
```
ref12-amd64:~% cat /tmp/test.sh 
#!/bin/sh

set -xe

cleanup() {
	set +e
	false
}

trap cleanup INT TERM EXIT
echo OK
ref12-amd64:~% /tmp/test.sh    
+ trap cleanup INT TERM EXIT
+ echo OK
OK
+ cleanup
+ set +e
+ false
ref12-amd64:~% echo $?
0
```
That is the return value of cleanup did not change the script exit status. Do you observe different?

I understand your sentiment about not enough time, but I don't think it is a good idea to add "just in case" plugs to the test script, as it will make it more confusing, hide some potential issues and will make debug real problems harder.

Reply:
@trociny 
I fully agree to not cutting corners.
I'll do it in another way, and make it my private source problem: I'll keep this as a local patch and apply this to the tree in my jenkins testing, until I've figured it out why I messed up here.
HEAD in itself is then clean.